### ISSUE TITLE
High CPU usage (pegs at 50%) on idle - Noobgamer0111

### ISSUE THREAD

**Issue by Noobgamer0111** [philippe44/LMS-uPnP#60]
Dear Phillipe-44,

Thank you for making this bridge so I can control my Sonos players via UPnP. I was previously using the Plex for Sonos plug-in for Plex Media Server but it was transcoding my precious FLACs to 296kbps AAC for no good reason.

Issue: Very high CPU usage for squeeze2upnp-win.exe at 50%, despite nothing playing (idle).

Specs: Dell Inspiron 15R 7520 SE (laptop from 2012)
CPU: Intel Core i7-3612QM @ 2.10GHz
RAM: 2x 4GB DDR3 @ 1600MHz
GPU: Intel HD Graphics 4000
SSD (boot drive, location of LMS install): Crucial MX500 SSD 1TB (CT1000MX500SSD1)
HDD (external USB3.0 drive, Music folder location): Seagate Backup Plus Slim Portable Drive 4TB (SRD00F1)

OS: Windows 10 Home 21H2 (OS Build 19044.1503)
LMS version: 8.2.0 - 1627922070 @ Tue Aug 3 11:28:42 WEDT 2021 (stable).
UPnP/DLNA Bridge version: 1.82.1 (acquired from the plug-ins page on the LMS Web controller).

Clients:
Sonos Playbar (S9), and 2x Sonos Amp (S16). All three devices are running Sonos S2 firmware v14.0 (build 67125031, stable).
Documentation:
https://www.sonos.com/support/en-us/sonos-user-guide/index.html#t=sonos-user-guide%2Fsonos-user-guide%2Fsonos-user-guide.htm
https://developer.sonos.com/build/content-service-add-features/supported-audio-formats/
https://developer.sonos.com/build/content-service-add-features/supported-audio-formats/flac-best-practices/ (slightly out of date, S2 now supports 24-bit FLACs and ALACs)

Images:
![Screenshot 2022-02-09 122138](https://user-images.githubusercontent.com/60173797/153104021-3bad0686-b2c4-40db-b4ba-f34438df16a0.png)
(Shows my current settings, I wish there was a button to reset settings to default without uninstalling the plug-in).

What I did:

- Played "How do I make you love me" by the Weeknd from "Dawn FM" (Qobuz rip, @ 48kHz-24bit 2.0 FLAC). Played at correct bit-rate and depth according to MediaInfo and Foobar2000.
- Saw this plug-in and screwed around with it (no playback was occuring when I was messing around).
- Noticed that the CPU usage was being pegged at near 50%, this was the only program I saw in Task Manager at this usage.
- Disabled the program in the Advanced tab of the Web controller. Until it is patched, this is the only effective solution to reduce CPU usage (and the heat).

**Comment by Noobgamer0111** [comment]
> Did you check older threads?
Yes, I did look up the "*.exe high cpu usage" on Google Chrome, but I could not figure what was the actual solution or fix made.

**Comment by philippe44** [comment]
can you post a log (clean the current one, don't set any additional debug and restart the bridge)

**Comment by Noobgamer0111** [comment]
Sure, I will send it to your outlook.com email. It will come from gmail.com.

**Comment by Noobgamer0111** [comment]
@philippe44 Any updates?

**Comment by philippe44** [comment]
Not really - The log is full of issues and warnings from different places but I missed the start of squeeze2upnp so I could not see if the parameters were correct. The only reason I know for 100% is missing the -z or -Z command, and this is what I was looking for. Maybe try to take the *bridge* log then and post it (clean log before)

**Comment by Noobgamer0111** [comment]
Sorry for the late reply.

Here's the new logs:
[server.log - 16.02.2022.zip](https://github.com/philippe44/LMS-to-uPnP/files/8076930/server.log.-.16.02.2022.zip)

**Comment by philippe44** [comment]
No what I nee dis the log of the bridge, not the log of LMS - you can find it in the bridge's settings

**Comment by Noobgamer0111** [comment]
I've decided to copy and paste the logs into a txt file, since I don't know how to extract them easily.
I've set Additional debugging to "All".
[UPnP.DLNA Bridge Logs - 20.02.2022 11.12.23.txt](https://github.com/philippe44/LMS-to-uPnP/files/8103224/UPnP.DLNA.Bridge.Logs.-.20.02.2022.11.12.23.txt)

Not sure if anything else is needed.
I'm including the config.xml as well.
[upnpbridge-config.txt](https://github.com/philippe44/LMS-to-uPnP/files/8103227/upnpbridge-config.txt)
